extensions/src/error/PersistenceError.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Error thrown when trying to write MSAL cache to persistence.
+ */
+export class PersistenceError extends Error {
+
+    // Short string denoting error
+    errorCode: string;
+    // Detailed description of error
+    errorMessage: string;
+
+    constructor(errorCode: string, errorMessage: string) {
+        const errorString = errorMessage ? `${errorCode}: ${errorMessage}` : errorCode;
+        super(errorString);
+        Object.setPrototypeOf(this, PersistenceError.prototype);
+
+        this.errorCode = errorCode;
+        this.errorMessage = errorMessage;
+        this.name = "PersistenceError";
+    }
+
+    static createFileSystemError(errorCode: string, errorMessage: string): PersistenceError {
```

Comment:
nitpick but can we add comments to the errors here?

Reply:
Sure, will add.